model/flow/sealing_segment.go

```diff
@@ -413,6 +385,53 @@ func (builder *SealingSegmentBuilder) addExecutionResult(result *ExecutionResult
 // Errors expected during normal operation:
 //   - InvalidSealingSegmentError if the added block would cause an invalid resulting segment
 func (builder *SealingSegmentBuilder) SealingSegment() (*SealingSegment, error) {
+
+	// at this point, go through all blocks and store any results which are referenced
+	// by blocks in the segment, but not contained within any blocks in the segment
+	missingExecutionResultMap := make(map[Identifier]struct{})
+
+	if builder.firstSeal != nil {
+		_, ok := builder.includedResults[builder.firstSeal.ResultID]
+		if !ok {
+			missingExecutionResultMap[builder.firstSeal.ResultID] = struct{}{}
+		}
+	}
+
+	for _, block := range append(builder.blocks, builder.extraBlocks...) {
+		for _, receipt := range block.Payload.Receipts {
+			_, included := builder.includedResults[receipt.ResultID]
+			if included {
+				continue
+			}
+			missingExecutionResultMap[receipt.ResultID] = struct{}{}
+		}
+		for _, seal := range block.Payload.Seals {
+			_, included := builder.includedResults[seal.ResultID]
+			if included {
+				continue
+			}
+			missingExecutionResultMap[seal.ResultID] = struct{}{}
+		}
+	}
+
+	// sort execution results to canonical order for consistent serialization
+	missingExecutionResults := make([]Identifier, 0, len(missingExecutionResultMap))
+	for resultID := range missingExecutionResultMap {
+		missingExecutionResults = append(missingExecutionResults, resultID)
+	}
+	slices.SortFunc(missingExecutionResults, func(a, b Identifier) int {
```

Comment:
Do we rely on this order somewhere? Don't quite understand why we need to sort them by identifier.